src/sections/user/user-edit-modal.tsx

```diff
@@ -0,0 +1,251 @@
+// user-edit-modal.tsx
+import React, { useState, useEffect } from 'react';
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+  Snackbar,
+  Alert,
+} from '@mui/material';
+
+type UserEditModalProps = {
+  open: boolean;
+  onClose: () => void;
+  user: {
+    id: string;
+    name: string;
+    company: string;
+    role: string;
+    status: string;
+    isVerified: boolean;
+    // avatarUrl: string;
+  };
+  onEditUser: (updatedUser: {
+    id: string;
+    name: string;
+    company: string;
+    role: string;
+    isVerified: boolean;
+    status: string;
+    // avatarUrl: string;
+  }) => void;
+};
+
+export function UserEditModal({ open, onClose, user, onEditUser }: UserEditModalProps) {
+  const [name, setName] = useState(user.name);
+  const [company, setCompany] = useState(user.company);
+  const [role, setRole] = useState(user.role);
+  const [status, setStatus] = useState(user.status);
+
+  const [errors, setErrors] = useState({
+    name: false,
+    company: false,
+    role: false,
+    status: false,
+  });
+
+  const [snackbarOpen, setSnackbarOpen] = useState(false); // Snackbar visibility state
+  const [snackbarMessage, setSnackbarMessage] = useState(''); // Snackbar message
+  const [snackbarSeverity, setSnackbarSeverity] = useState<'success' | 'error'>('success'); // Snackbar severity
```

Comment:
You can remove these snackbar implementation,
As I have added snackbar globally in updated code
You can take reference from ```src/sections/auth/auth-view.tsx```